sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt

```diff
@@ -152,19 +154,29 @@ class AppSyncRepository @Inject constructor(
         syncStore.clearAll(keepRecoveryCode = false)
     }
 
-    override fun logout(): Result<Boolean> {
+    override fun logout(deviceId: String): Result<Boolean> {
         val token = syncStore.token.takeUnless { it.isNullOrEmpty() }
             ?: return Result.Error(reason = "Token Empty")
-        val deviceId = syncStore.deviceId.takeUnless { it.isNullOrEmpty() }
-            ?: return Result.Error(reason = "Device Id Empty")
+
+        val logoutThisDevice = deviceId.isEmpty() || deviceId == syncStore.deviceId
+
+        val deviceId = if (logoutThisDevice) {
```

Comment:
deviceId is either `syncStore.deviceId` or we return Result.Error right? You can simplify all that logic here and also enforce `deviceId` to not be empty when passed into this method.

Reply:
it can be syncStore.deviceId or another deviceId (logging out a remote device)